scripts/mbedtls_framework/c_wrapper_generator.py

```diff
@@ -31,9 +30,16 @@ def c_declare(prefix: str, name: str, suffix: str) -> str:
     ('wrapper_name', str),
 ])
 
-def blockstr_format(in_str: str, new_lines: int = 1) -> str:
-    """Return a whitespace stripped str, appended with user defined empty lines."""
-    return inspect.cleandoc(in_str) + ("\n" * (new_lines + 1))
+def strip_indentation(in_str: str, new_lines: int = 1, ident_lv: int = 4) -> str:
```

Comment:
Why is the default `indent_lv` 4? Also, I'm not sure it makes sense to specify how much indentation to strip, given that the intent is that the input's indentation is based on the indentation of the surrounding code, while the output's indentation is based on the position in the generated code. The input's indentation can be determined by stripping the common whitespace sequence from the beginning of lines (line `dedent` and `cleandoc`). Let the user specify the desired output indentation: 0 for code outside a function, 1 for code in a function, 2 for code in a block in a function, etc.

Reply:
> Why is the default indent_lv 4? 

Pure statistical coincident. We use it most with level 4.

I have considered making it read the first line of code and automatically determine the level in 4 spaces segments. But giving the control of the filter to the caller would allow creating snippets like that. ( which are addmittely not very usefull )

~~~~~
        some_function();
     some_less_indented_code();
~~~~~

Would that be acceptable?

- Automatically remove indentation.
- Change the indent_lv parameter to adjust the output indendation?

Reply:
> Automatically remove indentation.
> Change the indent_lv parameter to adjust the output indendation?

Yes, I think that's the best choice if we don't preserve the source indentation.